### DISCUSSION TITLE
How to use them together: MS AutoGen and MS Semantic Kernel

### DISCUSSION THREAD

**Discussion by geoffreya** [microsoft/semantic-kernel#3003] [Q&A]
AutoGen is for integrating and controlling multiple LLMs. SK is for a single LLM. I get that.  OK but...

Might be nice to see where these two nice LLM libraries by MSFT overlap in features and where they don't and when to use. 

Like, maybe AutoGen is a superset and delegates to SK for using individual LLMs. I'm expecting this is not acctually true yet however.

If no thought was given to it yet, it would make sense because both libs are very new, and are still being developed. 

But I'm learning both because I can see use cases for both when I consider the apps I want to make.

Regards,

Geoffrey

**Comment by isaackhazi** [comment]
Good question! I'm curious to see how these libraries can be used together. For example we have already built a few bots with skills(plugins) that work for specific use cases. Rather than rebuild those skills as functions within AutoGen, would be great if AutGen agents can communicate with agents built using SK that already have a large number of skills. I suppose one way would be to expose the SK agent via an endpoint and create a function in AutoGen to call the end point with the relevant parameters.

**Comment by samleegithub** [comment]
https://devblogs.microsoft.com/semantic-kernel/autogen-agents-meet-semantic-kernel/

**Comment by samleegithub** [comment]
https://devblogs.microsoft.com/semantic-kernel/assistants-a-first-look-into-using-openai-assistants-with-semantic-kernel/

New SK capabilities that have been inspired by AutoGen ideas.

**Comment by unaihuete93** [comment]
Bringing back the question, as now Semantic Kernel also offers multi agent collaboration https://learn.microsoft.com/en-us/semantic-kernel/frameworks/agent/agent-chat?pivots=programming-language-csharp

**Comment by thangchung** [comment]
I just found out the plan for two-of-them in the new year https://devblogs.microsoft.com/autogen/microsofts-agentic-frameworks-autogen-and-semantic-kernel/
